src/pages/Teams.tsx

```diff
@@ -4,41 +4,57 @@ import {getTeams as fetchTeams} from '../api';
 import Header from '../components/Header';
 import List from '../components/List';
 import {Container} from '../components/GlobalComponents';
+import SearchBar from '../components/SearchBar';
 
-var MapT = (teams: TeamsList[]) => {
+const mapTeamsToItems = (teams: TeamsList[]) => {
     return teams.map(team => {
-        var columns = [
-            {
-                key: 'Name',
-                value: team.name,
-            },
-        ];
         return {
             id: team.id,
             url: `/team/${team.id}`,
-            columns,
+            columns: [
+                {
+                    key: 'Name',
+                    value: team.name,
+                },
+            ],
             navigationProps: team,
         } as ListItem;
     });
 };
 
 const Teams = () => {
-    const [teams, setTeams] = React.useState<any>([]);
-    const [isLoading, setIsLoading] = React.useState<any>(true);
+    const [teams, setTeams] = React.useState<TeamsList[]>([]);
+    const [isLoading, setIsLoading] = React.useState<boolean>(true);
+    const [error, setError] = React.useState<string | null>(null);
+    const [searchTerm, setSearchTerm] = React.useState<string>('');
+
+    const filteredTeams = teams.filter(team =>
+        team.name.toLowerCase().includes(searchTerm.toLowerCase())
+    );
 
     React.useEffect(() => {
-        const getTeams = async () => {
-            const response = await fetchTeams();
-            setTeams(response);
-            setIsLoading(false);
+        const fetchAndSetTeamsData = async () => {
+            try {
+                const teamData = await fetchTeams();
+                setTeams(teamData);
+            } catch (err) {
+                setError('Failed to fetch teams, please reload the page to retry');
+            } finally {
+                setIsLoading(false);
+            }
         };
-        getTeams();
+        fetchAndSetTeamsData();
     }, []);
 
+    if (error) {
+        return <div>{error}</div>;
+    }
+
     return (
         <Container>
-            <Header title="Teams" showBackButton={false} />
-            <List items={MapT(teams)} isLoading={isLoading} />
+            <Header title="Teams" showBackButton={false}/>
+            <SearchBar searchTerm={searchTerm} setSearchTerm={setSearchTerm} isLoading={isLoading} foundUsers={filteredTeams.length > 0} />
```

Comment:
Adding search bar with filtered teams length as the boolean, since its the only condition on this one